refactoring the convert code to pulumi page (#3921)
pulumi/pulumi-hugo: themes/default/content/docs/using-pulumi/adopting-pulumi/converters.md

---
title_tag: "Convert Code to Pulumi"

> meta_desc: Pulumi convert allows you to convert ARM, CloudFormation, Kubernetes CustomResources, Kubernetes YAML, and Terraform to Pulumi.

Reviewer: Nit, I think: I'm not sure that this is more accurate. The converters are what do the work, and this is not the reference for the `pulumi convert` command.

- /docs/converters/
---

> `pulumi convert` is a subcommand within the Pulumi Command Line Interface (CLI) you can use to convert between different source IaC languages or formats into all of Pulumi’s [supported languages](/docs/languages-sdks/), such as TypeScript, JavaScript, Go, Python, C#, or Java.

Reviewer 2: ```suggestion
`pulumi convert` is a subcommand within the Pulumi Command Line Interface (CLI) you can use to convert between different source Infrastructure as Code (IaC) languages or formats into all of Pulumi’s [supported languages](/docs/languages-sdks/), such as TypeScript, JavaScript, Go, Python, C#, or Java.
```

> `pulumi convert` is a subcommand within the Pulumi Command Line Interface (CLI) you can use to convert between different source Infrastructure as Code (IaC) languages or formats into all of Pulumi’s [supported languages](/docs/languages-sdks/), such as TypeScript, JavaScript, Go, Python, C#, or Java.

Reviewer: ```suggestion
The `pulumi convert` command allows you translate supported sources to the various [languages that Pulumi supports]((/docs/languages-sdks/), such as TypeScript, JavaScript, Go, Python, C#, or Java.). Conversion sources are supported via a plugin model, and include other IaC tools like Terraform. Pulumi YAML programs can also be converted to any other supported Pulumi language.
```

For the detailed usage of this command and options, refer to the [pulumi convert CLI documentation.](https://www.pulumi.com/docs/cli/commands/pulumi_convert/)

> ### Supported source languages

Reviewer 2: Shouldn't CloudFormation be in this list also?

> ### Supported source languages

Reviewer: This isn't really accurate because only Terraform and YAML are sources for the `pulumi convert` command. We also have legacy standalone conversion utilities like `arm2pulumi`, `cf2pulumi`, etc. We either need to change the scope of this page to be "conversion" in the general sense and include everything, or a `pulumi convert` usage guide and narrow the list of utilities to only those that are proper `pulumi convert` plugins.

It seems like the former is closer to the author's intent, so we should explain the "old vs. new" situation we have right now. I think the user cares a lot less about standalone executables versus a plugin model than they do "I have CloudFormation today. What are my options?"

* [YAML](/docs/languages-sdks/yaml/)
* [Terraform](/docs/using-pulumi/adopting-pulumi/migrating-to-pulumi/from-terraform/)
* [Kubernetes](/docs/using-pulumi/adopting-pulumi/migrating-to-pulumi/from-kubernetes/)
* [ARM](/docs/using-pulumi/adopting-pulumi/migrating-to-pulumi/from-azure)
* [Bicep](/docs/using-pulumi/adopting-pulumi/migrating-to-pulumi/from-azure)

### Supported destination languages

* [TypeScript/Javascript](/docs/languages-sdks/javascript/)
* [Go](/docs/languages-sdks/go/)
* [Python](/docs/languages-sdks/python/)
* [C#](/docs/languages-sdks/dotnet/)
* [Java](/docs/languages-sdks/java/)

### Use cases for using pulumi convert

> The `pulumi convert` subcommand is designed to address a variety of migration and conversion scenarios where infrastructure as code (IaC) practices evolve alongside your project needs. Here are some common use cases:

Reviewer 2: ```suggestion
The `pulumi convert` subcommand is designed to address a variety of migration and conversion scenarios where IaC practices evolve alongside your project needs. Here are some common use cases:
```

* **Migrating from other IaC tools:** If you're looking to transition your infrastructure definitions from tools like Terraform, Bicep, or ARM templates into Pulumi.

> * **Converting from Pulumi YAML:** If you started your project with Pulumi YAML and find your requirements have evolved, you can convert and take advantage of the benefits offered by popular programming languages, such as complex logic capabilities and the flexibility to create modular and reusable components.

Reviewer 2: Re: the "modular and resuable components" part, is this a reference to [Pulumi Components](https://www.pulumi.com/docs/concepts/resources/components/)? If so, should we link it here?
